streams/src/test/java/org/apache/kafka/streams/processor/internals/RecordCollectorTest.java

```diff
@@ -126,10 +129,10 @@ public void setup() {
         clientSupplier.setCluster(cluster);
         streamsProducer = new StreamsProducer(
             config,
-            "threadId",
+            processId + "-StreamThread-1",
```

Comment:
Just hard code `"-StreamThread-1"` -- why do we need to prefix it? We used a hard-coded value before.

Reply:
StreamsProducer tries to parse the thread name to get the `processId` for the null check that's only done for eos-v2 (see above)

Reply:
This PR does not change `StreamsProducer` so this parsing should have happened for `_beta` already -- what do I miss?